### ISSUE TITLE
Revert renaming Plans to Prices (feature or fix request)

### ISSUE THREAD

**Issue by top-master** [laravel/cashier-stripe#1341]
**See** [[13.x] Rename plans to prices #1166](https://github.com/laravel/cashier-stripe/pull/1166)

**As Web-Hooks prove:**

**1** Each time during a Stripe `Subscription` creation, if a Stripe-`Price`-object was added intto "items" instead of a Stripe-`Plan`-object, then Stripe automatically creates a `Plan` (in their DB), and that one for each `Price` added.

**2** Also, each time we create a Stripe `Plan` manually, Stripe automatically creates a Stripe `Price` (in their DB), to store things like `amount` and `currency`, and auto-deletes that `Price` later (once `Plan` is deleted).

**3** But **the *huge* difference is that**:
* A `Plan` forces us to set "`interval`" during creation (while "interval" is optional for `Price` objects), hence `Plan`-object is optimal for subscriptions.
* A `Plan` can have different `amount` then the `Product`'s `Price`.
* Let's consider we have one "newspaper" as Product, and we sell a single newspaper by a higher *`Price`* (if people have no subscription), then which of below makes no sense:
    * "Our Subscription *`Plan`* has separate lower `Price`."
    * "Our Subscription *`Price`* has separate lower `Price`."

In other words, each `Subscription`-item needs both a `Plan` and a `Price`, and although Stripe may create requirements automatically, above proves it was a *huge* mistake to rename everything from `Plans` to `Prices` (IMHO).

If you want to clarify the difference of `Plans` and `Prices` to developers, then simply just do that, by writting somewhere:
>A `Plan` is the requirement to create `Subscriptions` (because it stores `interval`, `interval_count` and other settings), but if a `Price` has `recurring.interval` set, then Stripe will automatically create a `Plan` for us.

**Comment by driesvints** [comment]
From the Stripe API docs:

<img width="691" alt="Screenshot 2022-04-04 at 09 04 00" src="https://user-images.githubusercontent.com/594614/161490772-15d88298-d19a-48f7-bd07-dc67abbb154e.png">

So we've made the choice to fully move to the Prices API. We won't be reverting the related PR, sorry.

**Comment by top-master** [comment]
For further reference, the https://stripe.com/docs/billing/migration/migrating-prices URL says:
>"[Prices](https://stripe.com/docs/api/prices) API adds more ...  replaces the Plans API, so Stripe recommends migrating"

---

In other words it's Stripe's fault that, they force you to copy/paste the `Price` wherever a  Recurring-`Price` is required, **even if the Recurring-Price's "amount" is exactly the same as Non-Recurring-Price's.**

It's not Laravel-Cashier's fault that they followed Stripe's instructions.

---

I noticed that, each of the "items" of Stripe's "Subscription"-object do include both a "plan", and a "price",
looking forther, they both have exact same ID.
>```
>$item = $subscription['items']['data'][0];
>echo $item['price']['id'] === $item['plan']['id'];
>```
